Sources/Controllers/2.0/RoutePlanning/OAExitRoutePlanningBottomSheetViewController.m

```diff
@@ -61,22 +61,70 @@ - (void) applyLocalization
 
 - (CGFloat) initialHeight
 {
-    CGFloat width = DeviceScreenWidth - 2 * kHorizontalMargin;
+    CGFloat width;
+    if ([OAUtilities isLandscape])
+        width = OAUtilities.isIPad ? kOABottomSheetWidthIPad : kOABottomSheetWidth;
+    else
+        width = DeviceScreenWidth;
+    
     CGFloat headerHeight = self.headerView.frame.size.height;
-    CGFloat contentHeight = [OAUtilities calculateTextBounds:OALocalizedString(@"plan_route_exit_message") width:width font:[UIFont systemFontOfSize:15.]].height + _exitButton.frame.size.height + kVerticalMargin * 2;
-    CGFloat buttonsHeight = 132. + [OAUtilities getBottomMargin];
+    CGFloat contentHeight = [OAUtilities calculateTextBounds:OALocalizedString(@"plan_route_exit_message") width:width font:[UIFont systemFontOfSize:15.]].height;
+    contentHeight += _exitButton.frame.size.height + kVerticalMargin * 2;
+    CGFloat buttonsHeight = [self buttonsViewHeight];
     return headerHeight + contentHeight + buttonsHeight + kVerticalMargin * 2;
 }
 
 - (CGFloat) buttonsViewHeight
 {
-    CGFloat extraBottomOffset = [OAUtilities getBottomMargin] > 0 ? 0 : kVerticalMargin;
-    return kButttonsHeight + extraBottomOffset;
+    CGFloat bottomPadding = [OAUtilities getBottomMargin];
+    bottomPadding = bottomPadding == 0 ? kVerticalMargin : bottomPadding;
+    return kButtonsHeightWithoutBottomPadding + bottomPadding;
 }
 
 - (CGFloat) getViewHeight
 {
-    return self.initialHeight;
+    return [self initialHeight];
+}
+
+- (void) adjustFrame
+{
+    CGRect f = self.bottomSheetView.frame;
```

Comment:
Was it really necessary to expose the contentContainer and buttonsView to subclasses? I think the base class should handle the scroll behavior. Instead of fixing the scrolling behavior you should refactor the implementation of ExitRoutePlanningBottomSheet to a tableview (as intended by the supercalss)